### ISSUE TITLE
Ability to create the ring without any private key

### ISSUE THREAD

**Issue by h5law** [noot/ring-go#11]
## Overview

For our usecase, we already know each public key that will be put into the ring prior to creating it. If we had the ability to create the ring from the public keys without inserting another public key (corresponding to the private key provided) this would be very beneficial to us.

Currently the `NewKeyRingFromPublicKeys(curve types.Curve, pubkeys []types.Point, privkey types.Scalar, idx int) (*Ring, error)` is the closest to this desired use case but I propose the following:

```go
// NewFixedKeyRingFromPublicKeys takes public keys and a curve to create a ring
func NewFixedKeyRingFromPublicKeys(curve types.Curve, pubkeys []types.Point) (*Ring, error) {
	size := len(pubkeys)
	newRing := make([]types.Point, size)
	for i := 0; i < size; i++ {
		newRing[i] = pubkeys[i]
	}

	return &Ring{
		pubkeys: newRing,
		curve:   curve,
	}, nil
}
```

Of course the naming might need to be reworked but this method would allow for the generation of a "fixed" ring given a list of public key points

**Comment by noot** [comment]
@h5law hey, thanks for the suggestion. this makes sense to me. let me know if you plan to PR this!

**Comment by noot** [comment]
closed by #12
